### DISCUSSION TITLE
Any chance to make some built in text customizable in 11.1?

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#13201] [Q&A] [answered]
Example: date picker placeholder, managed file dialog button text.

**Comment by maxkatz6** [comment, marked as answer]
I going to take some ideas from https://github.com/AvaloniaUI/Avalonia/discussions/8741 or maybe IResourceHost based solution in the next month (before 11.1). Any, even non-perfect, solution would be better than no solution at all.

So far you can still do it, but with visual tree manipulations, directly editing controls.

**Reply by timunie** [reply]
I think the fastest way would be to define strings via DynamicResource. A Locaclizer build in having a markup extension with IntelliSense would be best, if that is even possible. But most likely not fast to develop.

**Reply by thevortexcloud** [reply]
A markup extension that can pull values out of a resource dictionary/other source is not too bad to make. But making it generic enough that everyone can use it and with full binding support is also not straightforward (which is likely a requirement as a lot of apps will change what they display based on some value or need to swap languages on the fly). Personally I think if Avalonia did go down that route it should just define some sort of interface (with possibly some simple default implementations, EG in memory, CSV, etc) which things can implement and then register as a source for string lookups. That way the specifics of how it should work is up to each developer.

**Reply by maxkatz6** [reply]
> A markup extension that can pull values out of a resource dictionary/other source is not too bad to make

Again, it's not a goal to implement multi-purpose localization system for everybody. This question is only about localization of build-in avalonia controls, and providing an API to redirect/override build-in localization.
